generators/client-2/files.js

```diff
@@ -0,0 +1,651 @@
+            type: 'copyJs',
+            from: ANGULAR_DIR,
+            templates: [
+                // entities
+                'entities/_entity.state.ts',
+                // home module
+                'home/_home.state.ts',
+                'layouts/error/_error.state.ts'
+            ]
+        },
+        {
+            type: 'copyHtml',
+            from: ANGULAR_DIR,
+            templates: [
+                // home module
+                'home/_home.html',
+                'layouts/navbar/_navbar.html',
+                'layouts/footer/_footer.html',
+                'layouts/error/_error.html'
+            ]
+        }
+    ],
+    image: [
+        {
+            type: 'copy',
+            from: MAIN_SRC_DIR,
+            templates: [
+                'content/images/_hipster.png',
+                'content/images/_hipster2x.png',
+                'content/images/_logo-jhipster.png'
+            ]
+        }
+    ],
+    clientTestFw: [
+        {
+            type: 'template',
+            from: TEST_SRC_DIR,
+            templates: [
+                '_karma.conf.js',
+                'spec/helpers/_module.js',
+                'spec/helpers/_httpBackend.js',
+                'spec/app/admin/health/_health.controller.spec.js',
+                'spec/app/account/password/_password.controller.spec.js',
+                'spec/app/account/password/_password-strength-bar.directive.spec.js',
+                'spec/app/account/settings/_settings.controller.spec.js',
+                'spec/app/account/activate/_activate.controller.spec.js',
+                'spec/app/account/register/_register.controller.spec.js',
+                'spec/app/account/reset/finish/_reset-finish.controller.spec.js',
+                'spec/app/account/reset/request/_reset-request.controller.spec.js',
+                'spec/app/services/auth/_auth.services.spec.js',
+                'spec/app/components/login/_login.controller.spec.js'
+            ]
+        },
+        {
+            type: 'template',
+            condition: generator => generator.authenticationType === 'session',
+            from: TEST_SRC_DIR,
+            templates: [
+                'spec/app/account/sessions/_sessions.controller.spec.js'
+            ]
+        },
+        {
+            type: 'template',
+            condition: generator => generator.testFrameworks.indexOf('protractor') !== -1,
+            from: TEST_SRC_DIR,
+            templates: [
+                'e2e/account/_account.js',
+                'e2e/admin/_administration.js',
+                '_protractor.conf.js'
+            ]
+        }
+    ]
+};
+
+module.exports = {
+    writeFiles,
+    files
+};
+
+function writeFiles() {
+    return {
+        writeSepcial: function () {
+            this.fs.copy(this.templatePath('gulp/_handle-errors.js'), this.destinationPath('gulp/handle-errors.js')); // to avoid interpolate errors
+            mkdirp(MAIN_SRC_DIR);
+        },
+
+        writeFiles: function () {
```

Comment:
this method is generic and hence can be moved to generator-base if we are adopting this. so the write block will reduce to just a single method call